traitsui/tests/editors/test_enum_editor.py

```diff
@@ -180,46 +374,199 @@ def check_enum_text_bad_update(self, view):
     def test_simple_evaluate_editor_bad_text(self):
-        self.check_enum_text_bad_update(simple_evaluate_view)
+        self.check_enum_text_bad_update(get_evaluate_view("simple"))
 
     @skip_if_null
     @unittest.skipIf(is_windows, "Test needs fixing on windows")
     def test_simple_evaluate_editor_object(self):
-        self.check_enum_object_update(simple_evaluate_view)
+        self.check_enum_object_update(get_evaluate_view("simple"))
 
     @skip_if_null
-    @unittest.skipIf(is_windows, "Test needs fixing on windows")
-    def test_simple_evaluate_popup_editor_text(self):
-        self.check_enum_text_update(simple_evaluate_view_popup)
+    def test_simple_evaluate_editor_object_no_auto_set(self):
+        view = get_evaluate_view("simple", auto_set=False)
+        enum_edit = EnumModel()
+
+        with store_exceptions_on_all_threads():
+            gui, combobox = self.setup_gui(enum_edit, view)
+
+            self.assertEqual(enum_edit.value, "one")
+
+            set_combobox_text(combobox, "two")
+            gui.process_events()
+
+            # wx modifies the value without the need to finish entry
+            if is_current_backend_qt4():
+                self.assertEqual(enum_edit.value, "one")
+
+                finish_combobox_text_entry(combobox)
+                gui.process_events()
+
+            self.assertEqual(enum_edit.value, "two")
 
     @skip_if_null
-    @unittest.skipIf(is_windows, "Test needs fixing on windows")
-    def test_simple_evaluate_popup_editor_index(self):
-        self.check_enum_index_update(simple_evaluate_view_popup)
+    def test_simple_editor_resizable(self):
+        # Smoke test for `qt4.enum_editor.SimpleEditor.set_size_policy`
+        enum_edit = EnumModel()
+        resizable_view = View(UItem("value", style="simple", resizable=True))
+
+        with store_exceptions_on_all_threads():
+            ui = enum_edit.edit_traits(view=resizable_view)
+            self.addCleanup(ui.dispose)
 
     @skip_if_null
-    def test_simple_evaluate_popup_editor_bad_text(self):
-        self.check_enum_text_bad_update(simple_evaluate_view_popup)
+    def test_simple_editor_rebuild_editor_evaluate(self):
+        # Smoke test for `wx.enum_editor.SimpleEditor.rebuild_editor`
+        enum_editor_factory = EnumEditor(
+            evaluate=True,
+            values=["one", "two", "three", "four"],
+        )
+        view = View(UItem("value", editor=enum_editor_factory, style="simple"))
+
+        with store_exceptions_on_all_threads():
+            gui, combobox = self.setup_gui(EnumModel(), view)
+
+            enum_editor_factory.values = ["one", "two", "three"]
+
+
+class TestRadioEnumEditor(unittest.TestCase):
+
+    def setup_gui(self, model, view):
+        gui = GUI()
+        ui = model.edit_traits(view=view)
+        self.addCleanup(ui.dispose)
+
+        gui.process_events()
+        editor = ui.get_editors("value")[0]
+        widget = editor.control
+
+        return gui, widget
```

Comment:
Just an observation: The `create_ui` context manager being introduced in https://github.com/enthought/traitsui/pull/664/files could be useful here.  I will try to review that PR soon.

Reply:
I'm not sure it would be that useful here because GUI for wx tests would still have to be set up and cleaned up manually.

Reply:
Yes, we are missing the GUITestAssistant for Wx too: https://github.com/enthought/pyface/pull/266

In any case, what I meant to say is that this is a pattern repeated so many times that it needs to be refactored. I was also meaning to reference an implementation where this use case is solved using a context manager.

Reply:
Yes, I agree about refactoring but I don't think there's a need to rush the review on https://github.com/enthought/traitsui/pull/664 because it wouldn't help here until the same exists for wx. And I don't think that's going to happen soon enough to make it into this PR.